mockgcp/mockcontainer/nodepool.go

```diff
@@ -0,0 +1,209 @@
+	}
+
+	if err := s.storage.Create(ctx, fqn, obj); err != nil {
+		return nil, status.Errorf(codes.Internal, "error creating nodepool: %v", err)
+	}
+
+	return s.newLRO(ctx, name.Project.ID)
+}
+
+func (s *ClusterManagerV1) populateNodePoolDefaults(obj *pb.NodePool) error {
+	obj.Status = pb.NodePool_RUNNING
+
+	if obj.Management == nil {
+		obj.Management = &pb.NodeManagement{
+			AutoUpgrade: true,
+			AutoRepair:  true,
+		}
+	}
+
+	if obj.MaxPodsConstraint == nil {
+		obj.MaxPodsConstraint = &pb.MaxPodsConstraint{
+			MaxPodsPerNode: 110,
+		}
+	}
+
+	return nil
+}
+
+func (s *ClusterManagerV1) UpdateNodePool(ctx context.Context, req *pb.UpdateNodePoolRequest) (*pb.Operation, error) {
+	reqName := req.GetName()
+
+	name, err := s.parseNodePoolName(reqName)
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+	obj := &pb.NodePool{}
+	if err := s.storage.Get(ctx, fqn, obj); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, status.Errorf(codes.NotFound, "nodePool %q not found", reqName)
+		}
+		return nil, status.Errorf(codes.Internal, "error reading clonodePooludNodePool: %v", err)
+	}
+
+	klog.Infof("UpdateNodePool %v", prototext.Format(req))
+
+	update := proto.Clone(req).(*pb.UpdateNodePoolRequest)
+	update.Name = ""
+
+	// TODO: Support more updates!
+
+	if !proto.Equal(update, &pb.UpdateNodePoolRequest{}) {
+		return nil, status.Errorf(codes.InvalidArgument, "update was not fully implemented UpdateNodePoolRequest=%v", prototext.Format(update))
+	}
+
+	// // Required. The update mask applies to the resource.
+	// paths := req.GetUpdateMask().GetPaths()
+	// if len(paths) == 0 {
+	// 	klog.Warningf("update_mask was not provided in request, should be required")
+	// }
+
+	// // TODO: Some sort of helper for fieldmask?
+	// for _, path := range paths {
+	// 	switch path {
+	// 	case "description":
+	// 		obj.Description = req.GetNodePool().GetDescription()
+	// 	case "labels":
+	// 		obj.Labels = req.GetNodePool().GetLabels()
+	// 	case "timeout":
+	// 		obj.Timeout = req.GetNodePool().GetTimeout()
+	// 	default:
+	// 		return nil, status.Errorf(codes.InvalidArgument, "update_mask path %q not valid", path)
+	// 	}
+	// }
+
+	if err := s.storage.Update(ctx, fqn, obj); err != nil {
+		return nil, status.Errorf(codes.Internal, "error updating cloudNodePool: %v", err)
+	}
+
+	return s.newLRO(ctx, name.Project.ID)
+}
+
+func (s *ClusterManagerV1) DeleteNodePool(ctx context.Context, req *pb.DeleteNodePoolRequest) (*pb.Operation, error) {
+	name, err := s.parseNodePoolName(req.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+
+	oldObj := &pb.NodePool{}
+	if err := s.storage.Delete(ctx, fqn, oldObj); err != nil {
+		klog.Infof("deleting nodePool %q: %v", fqn, err)
+		if apierrors.IsNotFound(err) {
+			return nil, status.Errorf(codes.NotFound, "nodePool %q not found", name)
+		} else {
+			return nil, status.Errorf(codes.Internal, "error deleting nodePool: %v", err)
```

Comment:
Was there a reason to not use %w?

Reply:
IIRC status.Errorf doesn't allow it.  I'll verify.

Reply:
Confirmed, it's just a wrapper on fmt.Sprintf: https://github.com/grpc/grpc-go/blob/v1.60.0/status/status.go#L64

Maybe I should be using `status.FromError`, but I think the problem is that we need to return a particular status code at the http layer, which I don't think `status.FromError` would do